sky/skypilot_config.py

```diff
@@ -475,18 +475,16 @@ def _compose_cli_config(cli_config: Optional[str],) -> config_utils.Config:
         return config_utils.Config()
 
     config_source = 'CLI'
-    maybe_config_path = os.path.expanduser(cli_config)
     try:
-        if os.path.isfile(maybe_config_path):
+        maybe_config_path = os.path.expanduser(cli_config[0])
+        if len(cli_config) == 1 and os.path.isfile(maybe_config_path):
```

Comment:
```suggestion
        if os.path.isfile(maybe_config_path):
            if len(cli_config) != 1:
                raise ValueError('Cannot use multiple --config flags with a config file.')
```
Just to make sure the error is more clear in this case.